pkg/osquery/tables/apple_silicon_security_policy/table_test.go

```diff
@@ -0,0 +1,194 @@
+It is possible to make your system security much weaker and therefore easier to compromise using this tool.
+This tool is not to be used in production environments.
+It is possible to render your system unbootable with this tool.
+It should only be used to understand how the security of Apple Silicon Macs works.
+Use at your own risk!
+
+
+Local policy for volume group 123:
+OS environment:
+Boot Args Filtering Status:  Enabled    (sip3): absent
+
+Local policy for volume group 456:
+OS environment:
+Boot Args Filtering Status:  Enabled    (sip3): absent`
+
+const failedSecondVol = `This utility is not meant for normal users or even sysadmins.
+It provides unabstracted access to capabilities which are normally handled for the user automatically when changing the security policy through GUIs such as the Startup Security Utility in macOS Recovery ("recoveryOS").
+It is possible to make your system security much weaker and therefore easier to compromise using this tool.
+This tool is not to be used in production environments.
+It is possible to render your system unbootable with this tool.
+It should only be used to understand how the security of Apple Silicon Macs works.
+Use at your own risk!
+
+
+Local policy for volume group 5D0D176D-E8CC-4E8B-815C-444A679390BD:
+OS environment:
+OS Type                                       : macOS
+
+Local policy:
+Security Domain                         (SDOM): 0x1
+Production Status                       (CPRO): 1
+Can't get local policy for Volume Group UUID 5EADF322-3C85-4DF8-8516-0B556700CD5E
+Failed to display local policy for volume group 5EADF322-3C85-4DF8-8516-0B556700CD5E`
+
+func Test_ParsePoliciesTable(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name           string
+		status         string
+		expectedResult map[string]interface{}
+		queryClause    []string
+	}{
+		{
+			name:           "one boot volume",
+			status:         oneBootVol,
+			expectedResult: map[string]interface{}{"1234567890": []map[string]interface{}{{"OS_Type": map[string]interface{}{"code": "", "mode": "", "value": "macOS"}}}},
+		},
+		{
+			name:           "multiple boot volumes",
+			status:         multipleBootVol,
+			expectedResult: map[string]interface{}{"123": []map[string]interface{}{{"Boot_Args_Filtering_Status": map[string]interface{}{"code": "sip3", "mode": "Enabled", "value": "absent"}}}, "456": []map[string]interface{}{{"Boot_Args_Filtering_Status": map[string]interface{}{"code": "sip3", "mode": "Enabled", "value": "absent"}}}},
+		},
+		{
+			name:           "failed second volumes",
+			status:         failedSecondVol,
+			expectedResult: map[string]interface{}{"5D0D176D-E8CC-4E8B-815C-444A679390BD": []map[string]interface{}{{"OS_Type": map[string]interface{}{"code": "", "mode": "", "value": "macOS"}}, {"Security_Domain": map[string]interface{}{"code": "SDOM", "mode": "", "value": "0x1"}}, {"Production_Status": map[string]interface{}{"code": "CPRO", "mode": "", "value": "1"}}}},
```

Comment:
Just thinking out lout ... should we return something when there is a failed volume? Is there any value in that?

Reply:
I was able to get a second macOS install working and now the bputil output is sane with no failed message - so I suppose this error case will occur if a bootable volume exists but the macOS installer has not yet finished the entire install (occurs after you are able to successfully boot into it). I don't know if there are other cases where an error may occur.

Is there value in reporting a "failed" volume with no associated boot policy information? Yes if those scenarios (like an incomplete secondary macOS install) are valuable to know about. No if it's unreliable and/or would lead to spurious check failures.